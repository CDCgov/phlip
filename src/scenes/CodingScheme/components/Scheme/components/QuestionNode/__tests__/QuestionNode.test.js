import React from 'react'
import { shallow } from 'enzyme'
import { QuestionNode } from '../index'

const props = {
  listIndex: 0,
  node: {
    text: 'la la la',
    hovering: false
  },
  lowerSiblingCounts: [0],
  path: [0],
  scaffoldBlockPxWidth: 50,
  treeIndex: 0,
  isDragging: false,
  isOver: false,
  didDrop: false,
  canModify: true,
  canDrag: true,
  connectDragPreview: preview => preview,
  connectDragSource: handle => handle
}

describe('CodingScheme -- Scheme -- QuestionNode', () => {
  test('should render correctly', () => {
    expect(shallow(<QuestionNode {...props} />)).toMatchSnapshot()
  })

  describe('Question hovering', () => {
    test('should call setHoveredStatus on mouse enter', () => {
      const wrapper = shallow(<QuestionNode {...props} />)
      const card = wrapper.find('WithStyles(CardContent)').dive()
      const spy = jest.spyOn(wrapper.instance(), 'setHoveredStatus')
      card.find('CardContent').simulate('mouseenter')
      card.update()
      expect(spy).toHaveBeenCalled()
    })

    test('should call turnOffHover on mouse leave', () => {
      const wrapper = shallow(<QuestionNode {...props} />)
      const card = wrapper.find('WithStyles(CardContent)').dive()
      const spy = jest.spyOn(wrapper.instance(), 'setHoveredStatus')
      card.find('CardContent').simulate('mouseleave')
      card.update()
      expect(spy).toHaveBeenCalled()
    })

    test('should display actions if hovering = true and canModify = true', () => {
      const wrapper = shallow(<QuestionNode {...props} node={{ text: 'la la la' }} canModify={true} />)

      wrapper.setState({
        hovered: true
      })
      const card = wrapper.find('WithStyles(CardContent)').dive()
      expect(card.find('CardContent').find('WithTheme(Button)')).toHaveLength(3)
    })

    test('should only display edit action if canModify = false and hovering = true', () => {
      const wrapper = shallow(<QuestionNode {...props} node={{ text: 'la la la' }} canModify={false} />)
      wrapper.setState({
        hovered: true
      })
      const card = wrapper.find('WithStyles(CardContent)').dive()
      expect(card.find('CardContent').find('WithTheme(Button)')).toHaveLength(1)
    })
  })
})