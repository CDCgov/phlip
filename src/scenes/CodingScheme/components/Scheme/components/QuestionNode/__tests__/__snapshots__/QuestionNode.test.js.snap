// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodingScheme -- Scheme -- QuestionNode should render correctly 1`] = `
<div
  style={
    Object {
      "left": 50,
    }
  }
  tabIndex={-1}
>
  <div>
    <div
      onDragEnd={[Function]}
      onDragStart={[Function]}
      tabIndex={-1}
    >
      <Card
        style={
          Object {
            "backgroundColor": "white",
            "border": "none",
            "opacity": 1,
            "padding": "5px 10px",
            "width": 830,
          }
        }
        tabIndex={-1}
      >
        <div
          draggable={true}
          role="listitem"
          tabIndex={0}
        >
          <div
            aria-grabbed={false}
            draggable={true}
            role="button"
            tabIndex={0}
          >
            <withStyles(Tooltip)
              placement="bottom"
              text="Drag to reorder"
            >
              <withTheme(Icon)
                color="black"
                size="24"
              >
                reorder
              </withTheme(Icon)>
            </withStyles(Tooltip)>
          </div>
          <withStyles(CardContent)
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flex": 1,
                "padding": 5,
              }
            }
          >
            <withStyles(Typography)
              component="h4"
              noWrap={true}
              style={
                Object {
                  "flex": 1,
                }
              }
              type="subheading"
            >
              la la la
            </withStyles(Typography)>
            <div>
              <withStyles(Tooltip)
                aria-label="Add child question"
                id="add-child-question-0"
                placement="left"
                text="Add child question"
              >
                <withTheme(Button)
                  aria-label="Add child question"
                  color="accent"
                  component={[Function]}
                  style={
                    Object {
                      "height": 40,
                      "marginRight": 10,
                      "minHeight": "unset",
                      "minWidth": "unset",
                      "padding": 0,
                      "width": 40,
                    }
                  }
                  to={
                    Object {
                      "pathname": "/project/undefined/coding-scheme/add",
                      "state": Object {
                        "canModify": true,
                        "parentDefined": Object {
                          "hovering": false,
                          "text": "la la la",
                        },
                        "path": Array [
                          0,
                        ],
                      },
                    }
                  }
                  value={
                    <withTheme(Icon)
                      color="white"
                    >
                      subdirectory_arrow_right
                    </withTheme(Icon)>
                  }
                />
              </withStyles(Tooltip)>
              <withStyles(Tooltip)
                aria-label="View and edit question"
                id="edit-question-0"
                placement="right"
                text="Edit question"
              >
                <withTheme(Button)
                  aria-label="Edit question"
                  color="accent"
                  component={[Function]}
                  style={
                    Object {
                      "height": 40,
                      "marginRight": 10,
                      "minHeight": "unset",
                      "minWidth": "unset",
                      "padding": 0,
                      "width": 40,
                    }
                  }
                  to={
                    Object {
                      "pathname": "/project/undefined/coding-scheme/edit/undefined",
                      "state": Object {
                        "canModify": true,
                        "path": Array [
                          0,
                        ],
                        "questionDefined": Object {
                          "hovering": false,
                          "text": "la la la",
                        },
                      },
                    }
                  }
                  value={
                    <withTheme(Icon)
                      color="white"
                    >
                      mode_edit
                    </withTheme(Icon)>
                  }
                />
              </withStyles(Tooltip)>
              <withStyles(Tooltip)
                aria-label="Delete question"
                id="delete-question-0"
                placement="right"
                text="Delete question"
              >
                <withTheme(Button)
                  aria-label="Delete question"
                  color="accent"
                  onClick={[Function]}
                  style={
                    Object {
                      "height": 40,
                      "marginRight": 10,
                      "minHeight": "unset",
                      "minWidth": "unset",
                      "padding": 0,
                      "width": 40,
                    }
                  }
                  value={
                    <withTheme(Icon)
                      color="white"
                    >
                      delete
                    </withTheme(Icon)>
                  }
                />
              </withStyles(Tooltip)>
            </div>
          </withStyles(CardContent)>
          <withStyles(CardActions)
            disableActionSpacing={true}
            style={
              Object {
                "padding": 0,
              }
            }
          >
            <div
              style={
                Object {
                  "flex": "1 1 auto",
                }
              }
            />
          </withStyles(CardActions)>
        </div>
      </Card>
    </div>
  </div>
</div>
`;
